jena-arq/src/test/java/org/apache/jena/riot/lang/TestTurtleTerms.java

```diff
@@ -237,6 +241,41 @@
     @Test public void turtle_152() { parse("[ a <y> ; a <z> ] . ") ; }
     @Test public void turtle_153() { parse("[ a <z>, <z1> ] . ") ; }
 
+    // fatal characters
+    @Test(expected=RiotException.class)
+    public void turtle_154() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_<_open_angle_bracket> . "); }
+    @Test(expected=RiotException.class)
+    public void turtle_155() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_>_close_angle_bracket> . "); }
+    @Test(expected=RiotException.class)
+    public void turtle_156() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_\\_backslash> . "); }
+    @Test(expected=RiotException.class)
+    public void turtle_157() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_\n_newline> . "); }
+    @Test(expected=RiotException.class)
+    public void turtle_158() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_\r_carriage_return> . "); }
+
+    // warning characters
+    @Test public void turtle_159() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_\t_tab> . "); }
+    @Test public void turtle_160() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_ _space> . "); }
+    @Test public void turtle_161() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_{_open_curly_brace> . "); }
+    @Test public void turtle_162() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_}_close_curly_brace> . "); }
+    @Test public void turtle_163() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_\"_double_quote> . "); }
+    @Test public void turtle_164() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_|_pipe> . "); }
+    @Test public void turtle_165() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_^_caret> . "); }
+    @Test public void turtle_166() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_`_backtick> . "); }
+    @Test public void turtle_167() { laxParse("<http://example/foo> <http://example/bar> <http://example/invalid/iri/with_\f_form_feede> . "); }
+
+    private static void laxParse(String string) {
+        PeekReader r = PeekReader.readString(string) ;
+        Tokenizer tokenizer = TokenizerText.create()
```

Comment:
Simpler: `TokenizerText.create().fromString(string)` - no need to create the `PeekReader`.

(Some or all of `TestTokenizer` predates `fromString`.)